### PR TITLE
feat: Preliminary work for grouped alignment

### PR DESCRIPTION
**Pull request by gagnonlg**
This PR contains preliminary work on the superstructure alignment mechanism. It contains a GeometryHiearchyList-based grouping mechanism courtesy of @LaraCalic and simple unit tests by me.

### CODE REVIEW

File: Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp [hunk 1]
```diff
@@ -24,6 +25,35 @@
 
 namespace ActsExamples {
 
+class AlignmentGroup {
```
Comment (paulgessinger) [inline Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp:28]: Ultimately this should go into core. I don't think it has any Examples dependency, so I think it should be an easy move.

Comment (paulgessinger) [reply Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp:28]: We can leave it like this for this PR. I don't think there is an alignment plugin at this point, and I'd probably advocate making this a subdirectory in Core directly.

File: Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp [hunk 2]
```diff
@@ -24,6 +25,35 @@
...
+ public:
+  AlignmentGroup(const std::string& name,
+                 const std::vector<Acts::GeometryIdentifier>& geoIds)
+      : m_name(name), m_map(constructHierarchyMap(geoIds)) {}
+
+  // Access the name of the group
+  std::string getNameOfGroup() const { return m_name; }
+
+  // Useful for testing
+  bool has(Acts::GeometryIdentifier geoId) {
```
Comment (paulgessinger) [inline Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp:38]:
```suggestion
  bool has(Acts::GeometryIdentifier geoId) const {
```

**Review by paulgessinger — APPROVED** [review]

File: Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp (hunk 1 above)
Comment (gagnonlg) [reply Examples/Algorithms/Alignment/include/ActsExamples/Alignment/AlignmentAlgorithm.hpp:28]: Sounds good to me!

**Comment by paulgessinger** [comment]
I added @LaraCalic to the CI allowlist now. Can either on of you push an empty commit to trigger the pipeline again?

**Comment by gagnonlg** [comment]
Done @paulgessinger  but you will need to reaprove!

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [c5cc63a1eaa0b0377a5aefa45ff8ba4099f31391]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35332366) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6745668)!

**Comment by paulgessinger** [comment]
Athena failure is unrelated to this PR.

**Pull request merged**